lib/Wikia/src/Tracer/RequestId.php

```diff
@@ -43,16 +43,21 @@ function getRequestId() {
 			return $this->requestId;
 		}
 
-		// try to get the value from request X-Request-Id header
+		// try to get the value from request's X-Request-Id header or env variable
 		// we can't simply use $wgRequest as it's not yet set at this point
 		// this method is called on WikiFactoryExecuteComplete hook
 		// TODO: handle X-Trace-Id header as well
 		$headerValue = !empty( $_SERVER['HTTP_X_REQUEST_ID'] ) ? $_SERVER['HTTP_X_REQUEST_ID'] : false;
+		$envValue = !empty( $_ENV['X_TRACE_ID'] ) ? $_ENV['X_TRACE_ID'] : false;
```

Comment:
Here and in the WikiaTracer. I'm wondering about means to prevent accidental name clash in various situations. How about adding a distinct prefix which is very unlikely to occur and would give us more certainty that won't be randomly defined by anything else?